nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/PutS3Object.java

```diff
@@ -200,15 +201,24 @@
             .displayName("Server Side Encryption")
             .description("Specifies the algorithm used for server side encryption.")
             .required(true)
-            .allowableValues(NO_SERVER_SIDE_ENCRYPTION, ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION)
+            .allowableValues(NO_SERVER_SIDE_ENCRYPTION, ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION, KMS_MANAGED)
             .defaultValue(NO_SERVER_SIDE_ENCRYPTION)
             .build();
 
+    public static final PropertyDescriptor AWS_KMS_KEY = new PropertyDescriptor.Builder()
+            .name("aws-kms-key")
+            .displayName("Server Side Encryption using KMS")
```

Comment:
I recommend the displayName be a bit more direct, this field is the alias of a KMS key, and it's going to be used for SSE.